test/cppsuite/test_harness/workload/thread_context.cpp

```diff
@@ -109,6 +109,19 @@ transaction_context::commit(const std::string &config)
 {
     WT_DECL_RET;
     testutil_assert(_in_txn && !_needs_rollback);
+
+    /*
+     * We may have set the commit ts to a value that is now older than the current stable timestamp,
+     * we need to rollback as it is an invalid transaction. TODO - Should we have this check here or
+     * in the different operations defined in the database_operation class?
+     */
+    if (_timestamp_manager->get_stable_ts() >
+      ((WT_SESSION_IMPL *)_session)->txn->commit_timestamp) {
```

Comment:
Should we use `__session_query_timestamp` instead of accessing the `commit_timestamp` field directly?

Reply:
I don't think we have to do anything here, the code is basically the same as in [__txn_query_timestamp](https://github.com/wiredtiger/wiredtiger/blob/develop/src/txn/txn_timestamp.c#L224):
```
txn = session->txn;
...
*tsp = txn->commit_timestamp;
```

Reply:
This check will begin failing as soon as the first_commit_timestamp is used instead of the commit_timestamp. Also casting the session to a WT_SESSION_IMPL outside the api is a big no-no. One option to track this would be to save the first commit timestamp used on the thread context and then check against the stable. However the stable can still move concurrently and then you'd receive EINVAL in commit. So I think we should just check for EINVAL and rollback as required until we can implement a proper solution.

Reply:
@jiechenbo, no need to check for the oldest ts as by definition it is older than the stable ts, I will add a comment for this!
@luke-pearson, I will try this approach.

Reply:
Remove this entirely as we are accepting EINVAL now until WT-9198 is done.

Reply:
Furthermore, we don't need to call rollback at the cpp layer as WT will take care of it internally.